libsolidity/formal/CHC.cpp

```diff
@@ -541,6 +541,57 @@ void CHC::endVisit(Return const& _return)
 	m_currentBlock = predicate(*returnGhost);
 }
 
+bool CHC::visit(TryStatement const& _tryStatement)
+{
+	FunctionCall const* externalCall = dynamic_cast<FunctionCall const*>(&_tryStatement.externalCall());
+	solAssert(externalCall && externalCall->annotation().tryCall, "");
+	solAssert(m_currentFunction, "");
+
+	auto tryHeaderBlock = createBlock(&_tryStatement, PredicateType::FunctionBlock, "try_header_");
+	auto tryAfterCallSuccessBlock = createBlock(&_tryStatement, PredicateType::FunctionBlock, "try_after_call_success_");
+	auto tryAfterCallFailBlock = createBlock(&_tryStatement, PredicateType::FunctionBlock, "try_after_call_fail_");
+	auto afterTryBlock = createBlock(&m_currentFunction->body(), PredicateType::FunctionBlock);
+
+	connectBlocks(m_currentBlock, predicate(*tryHeaderBlock));
+	setCurrentBlock(*tryHeaderBlock);
+	// fine-grained control over how the external call is visited
+	// cannot use "_tryStatement.externalCall().accept(*this);" since we need to insert the "connectBlocks"
+	// visit(*externalCall); currently there is nothing to do here, so it's commented out
+	externalCall->expression().accept(*this);
+	ASTNode::listAccept(externalCall->arguments(), *this);
```

Comment:
As @ekpyron suggested, the code below this comment could be simplified by removing blocks `tryAfterCallSuccessBlock` and `tryAfterCallFailBlock`, and linking directly `m_currentBlock` to each clauseBlocks[i]`, where
```
endVisit(*externalCall);
tryCatchAssignment
```
should be added for `i == 0`.

Reply:
I agree that they are redundant. I have removed them and updated the code and the comments.